centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1574,9 +1574,9 @@ objDataCompare(int data_type_1, void* data_ptr_1, int data_type_2, void* data_pt
 
 		    case DATA_T_MONEY:
 		        m = (pMoneyType)data_ptr_2;
-			if (m->WholePart > intval) cmp_value = -1;
-			else if (m->WholePart < intval) cmp_value = 1;
-			else cmp_value = m->FractionPart?-1:0;
+			if (m->Value/10000 > intval) cmp_value = -1;
+			else if (m->Value/10000 < intval) cmp_value = 1;
+			else cmp_value = m->Value%10000?-1:0;
```

Comment:
It would probably be simpler to just compare (intval * 10000) to m->Value, returning -1, 1, or 0 as appropriate.